src/build-data/makefile.in

```diff
@@ -59,6 +59,14 @@ distclean:
 install: %{install_targets}
 	$(PYTHON_EXE) $(SCRIPTS_DIR)/install.py --prefix="%{prefix}" --build-dir="%{build_dir}" --bindir=%{bindir} --libdir=%{libdir} --docdir=%{docdir} --includedir=%{includedir}
 
+check: tests
+%{if build_shared_lib}
+	$(PYTHON_EXE) $(SCRIPTS_DIR)/check.py --test-exe="%{test_exe}" --shared-lib="%{shared_lib_name}"
+%{endif}
+%{unless build_shared_lib}
+	$(PYTHON_EXE) $(SCRIPTS_DIR)/check.py --test-exe="%{test_exe}"
+%{endif}
```

Comment:
Instead of this, pass to `check.py` the build directory via `--build-dir` and then load `build_config.json` to get test exe, shared lib usage etc